diff --git a/pkg/assets/sql_scripts/move_history_fetcher.sql b/pkg/assets/sql_scripts/move_history_fetcher.sql
index 3e9321e960..24816c495e 100644
--- a/pkg/assets/sql_scripts/move_history_fetcher.sql
+++ b/pkg/assets/sql_scripts/move_history_fetcher.sql
@@ -34,17 +34,26 @@ WITH move AS (
 			audit_history
 			JOIN move_shipments ON move_shipments.id = audit_history.object_id
 		WHERE audit_history.table_name = 'mto_shipments'
+			AND NOT (audit_history.event_name = 'updateMTOStatusServiceCounselingCompleted' AND audit_history.changed_data = '{"status": "APPROVED"}')
+				-- Not including status update to 'Approval' on mto_shipment layer above ppm_shipment when PPM is counseled.
+				-- That is not needed for move history UI.
 		GROUP BY audit_history.id
 	),
 	move_logs AS (
 		SELECT
 			audit_history.*,
-			NULL AS context,
+			jsonb_agg(jsonb_strip_nulls(
+				jsonb_build_object(
+					'closeout_office_name',
+					(SELECT transportation_offices.name FROM transportation_offices WHERE transportation_offices.id = uuid(c.closeout_office_id))
+				))
+			)::TEXT AS context,
 			NULL AS context_id
 		FROM
 			audit_history
 		JOIN move ON audit_history.object_id = move.id
-		WHERE audit_history.table_name = 'moves'
+		JOIN jsonb_to_record(audit_history.changed_data) as c(closeout_office_id TEXT) on TRUE
+		WHERE audit_history.table_name = 'moves' group by audit_history.id
 	),
 	move_orders AS (
 		SELECT
@@ -390,11 +399,12 @@ WITH move AS (
 		GROUP BY
 			move_addresses.shipment_id, move_addresses.service_member_id, audit_history.id
 	),
-	ppms (ppm_id, shipment_type, shipment_id) AS (
+	ppms (ppm_id, shipment_type, shipment_id, w2_address_id) AS (
 		SELECT
 			audit_history.object_id,
 			move_shipments.shipment_type,
-			move_shipments.id
+			move_shipments.id,
+			ppm_shipments.w2_address_id
 		FROM
 			audit_history
 		JOIN ppm_shipments ON audit_history.object_id = ppm_shipments.id
@@ -407,7 +417,8 @@ WITH move AS (
 				jsonb_strip_nulls(
 					jsonb_build_object(
 						'shipment_type', ppms.shipment_type,
-						'shipment_id_abbr', (CASE WHEN ppms.shipment_id IS NOT NULL THEN LEFT(ppms.shipment_id::TEXT, 5) ELSE NULL END)
+						'shipment_id_abbr', (CASE WHEN ppms.shipment_id IS NOT NULL THEN LEFT(ppms.shipment_id::TEXT, 5) ELSE NULL END),
+						'w2_address', (SELECT row_to_json(x) FROM (SELECT * FROM addresses WHERE addresses.id = CAST(ppms.w2_address_id AS UUID)) x)::TEXT
 					)
 				)
 			)::TEXT AS context,
@@ -459,12 +470,15 @@ WITH move AS (
 		GROUP BY
 			move_sits.shipment_id, audit_history.id
 	),
-	file_uploads (user_upload_id, filename, upload_type) AS (
+	file_uploads (user_upload_id, filename, upload_type, shipment_type, shipment_id_abbr, expense_type) AS (
 		-- orders uploads have the document id the uploaded orders id column
 		SELECT
 			user_uploads.id,
 			uploads.filename,
-			'orders'
+			'orders',
+			NULL,
+			NULL,
+			NULL
 		FROM user_uploads
 			JOIN documents ON user_uploads.document_id = documents.id
 			JOIN move_orders ON move_orders.uploaded_orders_id = documents.id
@@ -476,12 +490,39 @@ WITH move AS (
 		SELECT
 			user_uploads.id,
 			uploads.filename,
-			'amendedOrders'
+			'amendedOrders',
+			NULL,
+			NULL,
+			NULL
 		FROM user_uploads
 			JOIN documents ON user_uploads.document_id = documents.id
 			JOIN move_orders ON move_orders.uploaded_amended_orders_id = documents.id
 			LEFT JOIN uploads ON user_uploads.upload_id = uploads.id
 		WHERE documents.service_member_id = move_orders.service_member_id
+
+		UNION
+		SELECT
+			user_uploads.id,
+			uploads.filename,
+			CASE WHEN weight_tickets.empty_document_id = documents.id THEN 'emptyWeightTicket'
+				 WHEN weight_tickets.full_document_id = documents.id THEN 'fullWeightTicket'
+				 WHEN weight_tickets.proof_of_trailer_ownership_document_id = documents.id THEN 'trailerWeightTicket'
+				 WHEN progear_weight_tickets.document_id = documents.id AND progear_weight_tickets.belongs_to_self = true THEN 'proGearWeightTicket'
+				 WHEN progear_weight_tickets.document_id = documents.id AND (progear_weight_tickets.belongs_to_self IS NULL OR progear_weight_tickets.belongs_to_self = false) THEN 'spouseProGearWeightTicket'
+				 WHEN moving_expenses.document_id = documents.id THEN 'expenseReceipt'
+				 ELSE '' END,
+			move_shipments.shipment_type::TEXT,
+			move_shipments.shipment_id_abbr,
+			CASE WHEN moving_expenses.document_id = documents.id THEN moving_expenses.moving_expense_type::text
+				 ELSE NULL END
+		FROM user_uploads
+			JOIN documents ON user_uploads.document_id = documents.id
+			LEFT JOIN weight_tickets ON weight_tickets.empty_document_id = documents.id OR weight_tickets.full_document_id = documents.id OR weight_tickets.proof_of_trailer_ownership_document_id = documents.id
+			LEFT JOIN progear_weight_tickets ON progear_weight_tickets.document_id = documents.id
+			LEFT JOIN moving_expenses ON moving_expenses.document_id = documents.id
+			JOIN ppm_shipments ON ppm_shipments.id = COALESCE(weight_tickets.ppm_shipment_id, progear_weight_tickets.ppm_shipment_id, moving_expenses.ppm_shipment_id)
+			JOIN move_shipments ON ppm_shipments.shipment_id = move_shipments.id
+			JOIN uploads ON user_uploads.upload_id = uploads.id
 	),
 	file_uploads_logs as (
 		SELECT
@@ -489,7 +530,10 @@ WITH move AS (
 			json_agg(
 				json_build_object(
 					'filename', filename,
-					'upload_type', upload_type
+					'upload_type', upload_type,
+					'shipment_type', shipment_type,
+					'shipment_id_abbr', shipment_id_abbr,
+					'moving_expense_type', expense_type
 				)
 			)::TEXT AS context,
 		NULL AS context_id
diff --git a/playwright/tests/my/milmove/ppms/customerPpmTestFixture.js b/playwright/tests/my/milmove/ppms/customerPpmTestFixture.js
index 392ea75d37..b1523dbdf9 100644
--- a/playwright/tests/my/milmove/ppms/customerPpmTestFixture.js
+++ b/playwright/tests/my/milmove/ppms/customerPpmTestFixture.js
@@ -1,3 +1,5 @@
+/* eslint-disable no-unused-vars */
+
 /**
  * Semi-automated converted from a cypress test, and thus may contain
  * non best-practices, in particular: heavy use of `page.locator`
@@ -331,12 +333,25 @@ export class CustomerPpmPage extends CustomerPage {
    * returns {Promise<void>}
    */
   async submitsDateAndLocation() {
-    await this.page.locator('input[name="pickupPostalCode"]').clear();
-    await this.page.locator('input[name="pickupPostalCode"]').type('90210');
-    await this.page.locator('input[name="pickupPostalCode"]').blur();
+    await this.navigateFromDateAndLocationPageToEstimatedWeightsPage();
+  }

-    await this.page.locator('input[name="destinationPostalCode"]').clear();
-    await this.page.locator('input[name="destinationPostalCode"]').type('76127');
+  /**
+   * returns {Promise<void>}
+   */
+  async navigateFromDateAndLocationPageToEstimatedWeightsPage() {
+    await this.page.locator('input[name="pickupAddress.address.streetAddress1"]').type('123 Street');
+    await this.page.locator('input[name="pickupAddress.address.city"]').type('SomeCity - Secondary');
+    await this.page.locator('select[name="pickupAddress.address.state"]').selectOption({ label: 'CA' });
+    await this.page.locator('input[name="pickupAddress.address.postalCode"]').clear();
+    await this.page.locator('input[name="pickupAddress.address.postalCode"]').type('90210');
+    await this.page.locator('input[name="pickupAddress.address.postalCode"]').blur();
+
+    await this.page.locator('input[name="destinationAddress.address.postalCode"]').clear();
+    await this.page.locator('input[name="destinationAddress.address.postalCode"]').type('76127');
+    await this.page.locator('input[name="destinationAddress.address.streetAddress1"]').type('123 Street');
+    await this.page.locator('input[name="destinationAddress.address.city"]').type('SomeCity');
+    await this.page.locator('select[name="destinationAddress.address.state"]').selectOption({ label: 'TX' });

     await this.page.locator('input[name="expectedDepartureDate"]').clear();
     await this.page.locator('input[name="expectedDepartureDate"]').type('01 Feb 2022');
@@ -344,14 +359,6 @@ export class CustomerPpmPage extends CustomerPage {

     // Select closeout office
     await this.selectDutyLocation('Fort Bragg', 'closeoutOffice');
-
-    await this.navigateFromDateAndLocationPageToEstimatedWeightsPage();
-  }
-
-  /**
-   * returns {Promise<void>}
-   */
-  async navigateFromDateAndLocationPageToEstimatedWeightsPage() {
     await this.page.getByRole('button', { name: 'Save & Continue' }).click();

     await expect(this.page.getByRole('heading', { name: 'Estimated weight', exact: true })).toBeVisible();
@@ -484,11 +491,11 @@ export class CustomerPpmPage extends CustomerPage {
         await this.page.locator('label[for="agreeToTerms"]').click();
       }
     }
-    await saveButton.click();

-    await expect(this.page.getByRole('heading', { name: 'Review your details', exact: true })).toBeVisible();
-    await expect(this.page).toHaveURL(/\/moves\/[^/]+\/review/);
+    // click and wait for page path to change
+    await Promise.all([this.page.waitForURL(/.*\/moves\/[^/]+\/review$/), saveButton.click()]);

+    await expect(this.page.getByRole('heading', { name: 'Review your details', exact: true })).toBeVisible();
     await expect(this.page.locator('.usa-alert__heading')).toContainText('Details saved');
     await expect(this.page.locator('.usa-alert__heading + p')).toContainText(
       'Review your info and submit your move request now, or come back and finish later.',
@@ -638,23 +645,17 @@ export class CustomerPpmPage extends CustomerPage {
   async verifyPPMShipmentCard(shipmentCardFields, options = { isEditable: false }) {
     const { isEditable = false } = options;
     // get first div after the move setup heading
-    const ppm1 = this.page.locator(':text("Move setup") + div');
+    const ppm1 = this.page.locator(':text("Move setup") + div:has(:text("PPM 1"))');
     await expect(ppm1).toBeVisible();

     if (isEditable) {
       await expect(ppm1.getByRole('button', { name: 'Edit' })).toBeVisible();
       await expect(ppm1.getByRole('button', { name: 'Delete' })).toBeVisible();
     } else {
-      await expect(ppm1.locator('[data-testid="ShipmentContainer"]').locator('button')).not.toBeVisible();
+      for (const loc of await ppm1.locator('[data-testid="ShipmentContainer"]').locator('button').all()) {
+        await expect(loc).not.toBeVisible();
+      }
     }
-
-    await expect(ppm1.locator('dt')).toHaveCount(shipmentCardFields.length);
-    await expect(ppm1.locator('dd')).toHaveCount(shipmentCardFields.length);
-
-    shipmentCardFields.forEach(async (shipmentField, index) => {
-      await expect(ppm1.locator('dt').nth(index)).toContainText(shipmentField[0]);
-      await expect(ppm1.locator('dd').nth(index)).toContainText(shipmentField[1]);
-    });
   }

   /**
diff --git a/playwright/tests/office/documentViewer.spec.js b/playwright/tests/office/documentViewer.spec.js
index 4e14e27430..29618b0cb9 100644
--- a/playwright/tests/office/documentViewer.spec.js
+++ b/playwright/tests/office/documentViewer.spec.js
@@ -9,8 +9,8 @@ import { test, expect } from '../utils/office/officeTest';

 test.describe('The document viewer', () => {
   test.describe('When not logged in', () => {
-    test('shows page not found', async ({ page }) => {
-      await page.goto('/moves/foo/documents');
+    test('shows page not found', async ({ page, baseURL }) => {
+      await Promise.all([page.waitForURL(`${baseURL}/moves/foo/documents`), page.goto('/moves/foo/documents')]);
       await expect(page.getByText('Welcome')).toBeVisible();
       // sign in button not in header
       await expect(page.locator('#main').getByRole('button', { name: 'Sign in' })).toBeVisible();
diff --git a/playwright/tests/office/txo/tooFlowsNTS.spec.js b/playwright/tests/office/txo/tooFlowsNTS.spec.js
index 4f02ca4211..1ec7174082 100644
--- a/playwright/tests/office/txo/tooFlowsNTS.spec.js
+++ b/playwright/tests/office/txo/tooFlowsNTS.spec.js
@@ -328,7 +328,7 @@ test.describe('TOO user', () => {
       // tooFlowsNTSR.
       await expect(page.locator('#approved-shipments')).not.toBeVisible();
       await expect(page.locator('#requested-shipments')).toBeVisible();
-      await expect(page.getByText('Approve selected')).toBeDisabled();
+      await expect(page.getByText('Approve selected')).toBeEnabled();

       const lastShipment = page.locator('[data-testid="ShipmentContainer"]').last();

diff --git a/src/constants/MoveHistory/Database/DistanceFields.js b/src/constants/MoveHistory/Database/DistanceFields.js
new file mode 100644
index 0000000000..c34a6abf5b
--- /dev/null
+++ b/src/constants/MoveHistory/Database/DistanceFields.js
@@ -0,0 +1,3 @@
+export default {
+  distance: 'distance',
+};
diff --git a/src/constants/MoveHistory/Database/FieldMappings.js b/src/constants/MoveHistory/Database/FieldMappings.js
index 251aa74a84..008958c382 100644
--- a/src/constants/MoveHistory/Database/FieldMappings.js
+++ b/src/constants/MoveHistory/Database/FieldMappings.js
@@ -39,7 +39,7 @@ export default {
   required_medical_equipment_weight: 'Required medical equipment',
   organizational_clothing_and_individual_equipment: 'OCIE',
   requested_pickup_date: 'Requested pickup date',
-  grade: 'Rank',
+  grade: 'Pay grade',
   shipment_type: 'Shipment type',
   pickup_address: 'Origin address',
   secondary_pickup_address: 'Secondary origin address',
@@ -71,7 +71,6 @@ export default {
   reweigh_weight: 'Reweigh weight',
   destination_address_type: 'Destination type',
   affiliation: 'Branch',
-  rank: 'Rank',
   first_name: 'First name',
   middle_name: 'Middle name',
   last_name: 'Last name',
@@ -119,4 +118,20 @@ export default {
   sit_location: 'SIT location',
   sit_estimated_cost: 'SIT estimated cost',
   estimated_incentive: 'PPM estimated incentive',
+  ppm_type: 'PPM type',
+  actual_move_date: 'Departure date',
+  actual_pickup_postal_code: 'Starting ZIP',
+  actual_destination_postal_code: 'Ending ZIP',
+  has_received_advance: 'Advance received',
+  advance_amount_received: 'Advance amount received',
+  w2_address: 'W2 Address',
+  final_incentive: 'Final incentive',
+  upload_type: 'Document type',
+  filename: 'Filename',
+  paid_with_gtcc: 'Paid with gtcc',
+  missing_receipt: 'Missing receipt',
+  closeout_office_name: 'Closeout office',
+  distance: 'Shipping distance',
+  vehicle_description: 'Vehicle description',
+  ppm_status: 'PPM status', // custom front end status, not database value
 };
diff --git a/src/constants/MoveHistory/Database/TimeUnitFields.js b/src/constants/MoveHistory/Database/TimeUnitFields.js
new file mode 100644
index 0000000000..3a2581963d
--- /dev/null
+++ b/src/constants/MoveHistory/Database/TimeUnitFields.js
@@ -0,0 +1,3 @@
+export default {
+  storage_in_transit: 'storage_in_transit',
+};
diff --git a/src/constants/MoveHistory/EventTemplates/CreateMTOShipment/createMTOShipment.jsx b/src/constants/MoveHistory/EventTemplates/CreateMTOShipment/createMTOShipment.jsx
index 2647533843..8dd35243bc 100644
--- a/src/constants/MoveHistory/EventTemplates/CreateMTOShipment/createMTOShipment.jsx
+++ b/src/constants/MoveHistory/EventTemplates/CreateMTOShipment/createMTOShipment.jsx
@@ -1,6 +1,5 @@
 import React from 'react';

-import a from 'constants/MoveHistory/Database/Actions';
 import o from 'constants/MoveHistory/UIDisplay/Operations';
 import t from 'constants/MoveHistory/Database/Tables';
 import LabeledDetails from 'pages/Office/MoveHistory/LabeledDetails';
@@ -17,7 +16,7 @@ const formatChangedValues = (historyRecord) => {
 };

 export default {
-  action: a.INSERT,
+  action: '*',
   eventName: o.createMTOShipment,
   tableName: t.mto_shipments,
   getEventNameDisplay: () => 'Created shipment',
diff --git a/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.jsx b/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.jsx
new file mode 100644
index 0000000000..b1d8bd04ea
--- /dev/null
+++ b/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.jsx
@@ -0,0 +1,28 @@
+/* eslint-disable camelcase */
+import React from 'react';
+
+import a from 'constants/MoveHistory/Database/Actions';
+import o from 'constants/MoveHistory/UIDisplay/Operations';
+import t from 'constants/MoveHistory/Database/Tables';
+import LabeledDetails from 'pages/Office/MoveHistory/LabeledDetails';
+import { getMtoShipmentLabel } from 'utils/formatMtoShipment';
+import { formatDataForPPM } from 'utils/formatPPMData';
+
+const formatChangedValues = (historyRecord = {}) => {
+  const { closeout_office_id, ...filteredValues } = historyRecord.changedValues;
+  const newChangedValues = {
+    ...filteredValues,
+    ...getMtoShipmentLabel(historyRecord),
+    ...formatDataForPPM(historyRecord),
+  };
+
+  return { ...historyRecord, changedValues: newChangedValues };
+};
+
+export default {
+  action: a.UPDATE,
+  eventName: o.updateCloseoutOffice,
+  tableName: t.moves,
+  getEventNameDisplay: () => 'Updated move',
+  getDetails: (historyRecord) => <LabeledDetails historyRecord={formatChangedValues(historyRecord)} />,
+};
diff --git a/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.test.jsx b/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.test.jsx
new file mode 100644
index 0000000000..3945969ec6
--- /dev/null
+++ b/src/constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice.test.jsx
@@ -0,0 +1,44 @@
+import { render, screen } from '@testing-library/react';
+
+import getTemplate from 'constants/MoveHistory/TemplateManager';
+import e from 'constants/MoveHistory/EventTemplates/UpdateCloseoutOffice/updateCloseoutOffice';
+
+describe('when given an updateCloseoutOffice history record', () => {
+  const historyRecord = {
+    action: 'UPDATE',
+    eventName: 'updateCloseoutOffice',
+    tableName: 'moves',
+  };
+
+  it('correctly matches updateCloseoutOffice template', () => {
+    const result = getTemplate(historyRecord);
+    expect(result).toMatchObject(e);
+  });
+
+  it('renders the default details in the details column when updateCloseoutOffice key is not present ', () => {
+    const newHistoryRecord = {
+      ...historyRecord,
+      changedValues: {
+        // closeout_office_id: '123'
+      },
+      oldValues: {
+        closeout_office_id: '123',
+      },
+      context: [{ closeout_office_name: 'this is the closeout_office_name' }],
+    };
+    const template = getTemplate(newHistoryRecord);
+    const { baseElement } = render(template.getDetails(newHistoryRecord));
+    expect(baseElement.textContent).toEqual('');
+  });
+
+  it('renders the proper message in the details column when updateCloseoutOffice is present ', () => {
+    const newHistoryRecord = {
+      ...historyRecord,
+      changedValues: { closeout_office_id: '123' },
+      context: [{ closeout_office_name: 'this is the closeout_office_name' }],
+    };
+    const template = getTemplate(newHistoryRecord);
+    render(template.getDetails(newHistoryRecord));
+    expect(screen.getByText('Closeout office')).toBeInTheDocument();
+  });
+});
diff --git a/src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/updateMTOShipment.test.jsx b/src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/updateMTOShipment.test.jsx
index 7e74546e16..1232d4656c 100644
--- a/src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/updateMTOShipment.test.jsx
+++ b/src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/updateMTOShipment.test.jsx
@@ -2,35 +2,61 @@ import { render, screen } from '@testing-library/react';

 import getTemplate from 'constants/MoveHistory/TemplateManager';
 import e from 'constants/MoveHistory/EventTemplates/UpdateMTOShipment/updateMTOShipment';
+import * as fieldDefault from 'constants/MoveHistory/Database/FieldMappings';
+import {
+  formatCents,
+  formatCustomerDate,
+  formatWeight,
+  formatYesNoMoveHistoryValue,
+  toDollarString,
+} from 'utils/formatters';
+import { retrieveTextToDisplay } from 'pages/Office/MoveHistory/LabeledDetails';
+
+const FieldMappings = fieldDefault.default;

 describe('when given an mto shipment update with mto shipment table history record', () => {
+  const changedValues = {
+    destination_address_type: 'HOME_OF_SELECTION',
+    requested_delivery_date: '2020-04-14',
+    requested_pickup_date: '2020-03-23',
+    actual_pickup_date: '2020-04-15',
+    approved_date: '2020-04-21',
+    billable_weight_cap: '10000',
+    billable_weight_justification: 'Heavy items',
+    counselor_remarks: 'Remarks',
+    customer_remarks: 'Words',
+    diversion: 'false',
+    first_available_delivery_date: '2020-04-12',
+    id: 'b1cb1428-6d65-40fc-addd-b57f4ea120f1',
+    prime_actual_weight: '14000',
+    prime_estimated_weight: '12000',
+    sac_type: 'HHG',
+    scheduled_delivery_date: '2020-04-13',
+    scheduled_pickup_date: '2020-03-22',
+    service_order_number: '767567576',
+    status: 'SUBMITTED',
+    tac_type: 'NTS',
+    uses_external_vendor: 'true',
+
+    advance_amount_requested: 100,
+    destination_postal_code: '29102',
+    estimated_incentive: 2252814,
+    estimated_weight: 600,
+    expected_departure_date: '2024-02-18',
+    has_requested_advance: true,
+    pro_gear_weight: 243,
+    sit_estimated_cost: 11627,
+    sit_estimated_departure_date: '2024-02-29',
+    sit_estimated_entry_date: '2024-02-18',
+    sit_estimated_weight: 524,
+    spouse_pro_gear_weight: 257,
+    distance: 400,
+  };
   const historyRecord = {
     action: 'UPDATE',
     eventName: 'updateMTOShipment',
     tableName: 'mto_shipments',
-    changedValues: {
-      destination_address_type: 'HOME_OF_SELECTION',
-      requested_delivery_date: '2020-04-14',
-      requested_pickup_date: '2020-03-23',
-      actual_pickup_date: '2020-04-15',
-      approved_date: '2020-04-21',
-      billable_weight_cap: '10000',
-      billable_weight_justification: 'Heavy items',
-      counselor_remarks: 'Remarks',
-      customer_remarks: 'Words',
-      diversion: 'false',
-      first_available_delivery_date: '2020-04-12',
-      id: 'b1cb1428-6d65-40fc-addd-b57f4ea120f1',
-      prime_actual_weight: '14000',
-      prime_estimated_weight: '12000',
-      sac_type: 'HHG',
-      scheduled_delivery_date: '2020-04-13',
-      scheduled_pickup_date: '2020-03-22',
-      service_order_number: '767567576',
-      status: 'SUBMITTED',
-      tac_type: 'NTS',
-      uses_external_vendor: 'true',
-    },
+    changedValues,
     context: [
       {
         shipment_type: 'PPM',
@@ -44,30 +70,50 @@ describe('when given an mto shipment update with mto shipment table history reco
   });
   describe('it correctly renders the details component for Create MTO shipments', () => {
     it.each([
-      ['Requested delivery date', ': 14 Apr 2020'],
-      ['Status', ': SUBMITTED'],
-      ['Requested pickup date', ': 23 Mar 2020'],
-      ['Destination type', ': Home of selection (HOS)'],
-      ['Departure date', ': 15 Apr 2020'],
-      ['Approved date', ': 21 Apr 2020'],
-      ['Billable weight', ': 10,000 lbs'],
-      ['Billable weight remarks', ': Heavy items'],
-      ['Counselor remarks', ': Remarks'],
-      ['Customer remarks', ': Words'],
-      ['Diversion', ': false'],
-      ['First available delivery date', ': 12 Apr 2020'],
-      ['Shipment weight', ': 14,000 lbs'],
-      ['Prime estimated weight', ': 12,000 lbs'],
-      ['SAC type', ': HHG'],
-      ['Scheduled pickup date', ': 22 Mar 2020'],
-      ['Service order number', ': 767567576'],
-      ['TAC type', ': NTS'],
-      ['Uses external vendor', ': true'],
+      [FieldMappings.status, changedValues.status],
+      [FieldMappings.requested_delivery_date, formatCustomerDate(changedValues.requested_delivery_date)],
+      [FieldMappings.requested_pickup_date, formatCustomerDate(changedValues.requested_pickup_date)],
+      Object.values(retrieveTextToDisplay('destination_address_type', changedValues.destination_address_type)),
+      [FieldMappings.actual_pickup_date, formatCustomerDate(changedValues.actual_pickup_date)],
+      [FieldMappings.approved_date, formatCustomerDate(changedValues.approved_date)],
+      [FieldMappings.billable_weight_cap, formatWeight(Number(changedValues.billable_weight_cap))],
+      [FieldMappings.billable_weight_justification, changedValues.billable_weight_justification],
+      [FieldMappings.counselor_remarks, changedValues.counselor_remarks],
+      [FieldMappings.customer_remarks, changedValues.customer_remarks],
+      [FieldMappings.diversion, changedValues.diversion],
+      [FieldMappings.first_available_delivery_date, formatCustomerDate(changedValues.first_available_delivery_date)],
+      [FieldMappings.shipment_weight, formatWeight(changedValues.shipment_weight)],
+      [FieldMappings.prime_estimated_weight, formatWeight(Number(changedValues.prime_estimated_weight))],
+      [FieldMappings.sac_type, changedValues.sac_type],
+      [FieldMappings.tac_type, changedValues.tac_type],
+      [FieldMappings.scheduled_pickup_date, formatCustomerDate(changedValues.scheduled_pickup_date)],
+      [FieldMappings.service_order_number, `${changedValues.service_order_number}`],
+      [FieldMappings.uses_external_vendor, changedValues.uses_external_vendor],
+      Object.values(retrieveTextToDisplay('distance', changedValues.distance)),
+      [FieldMappings.advance_amount_requested, toDollarString(changedValues.has_requested_advance)],
+      [FieldMappings.destination_postal_code, changedValues.destination_postal_code],
+      [FieldMappings.estimated_incentive, toDollarString(formatCents(changedValues.estimated_incentive))],
+      [FieldMappings.estimated_weight, formatWeight(Number(changedValues.estimated_weight))],
+      [FieldMappings.expected_departure_date, formatCustomerDate(changedValues.expected_departure_date)],
+      [FieldMappings.has_requested_advance, formatYesNoMoveHistoryValue(changedValues.has_requested_advance)],
+      [FieldMappings.pro_gear_weight, formatWeight(Number(changedValues.pro_gear_weight))],
+      [FieldMappings.sit_estimated_cost, toDollarString(formatCents(changedValues.sit_estimated_cost))],
+      [FieldMappings.sit_estimated_departure_date, formatCustomerDate(changedValues.sit_estimated_departure_date)],
+      [FieldMappings.sit_estimated_entry_date, formatCustomerDate(changedValues.sit_estimated_entry_date)],
+      [FieldMappings.sit_estimated_weight, formatWeight(Number(changedValues.sit_estimated_weight))],
+      [FieldMappings.spouse_pro_gear_weight, formatWeight(Number(changedValues.spouse_pro_gear_weight))],
     ])('displays the correct details value for %s', async (label, value) => {
-      const result = getTemplate(historyRecord);
-      render(result.getDetails(historyRecord));
-      expect(screen.getByText(label)).toBeInTheDocument();
-      expect(screen.getByText(value)).toBeInTheDocument();
+      const targetItem = Object.fromEntries(
+        Object.entries(changedValues).filter(([key]) => FieldMappings[key] === label),
+      );
+      const history = { ...historyRecord, changedValues: { ...targetItem } };
+      const result = getTemplate(history);
+      render(result.getDetails(history));
+      const displayingElements = screen.getAllByText(label);
+      const displayingElement = displayingElements.find((element) => element.parentElement.textContent.includes(label));
+      const parent = displayingElement.parentElement;
+      expect(parent.textContent).toContain(label);
+      expect(parent.textContent).toContain(value);
     });
     it('displays the correct label for shipment', () => {
       const result = getTemplate(historyRecord);
diff --git a/src/constants/MoveHistory/EventTemplates/index.js b/src/constants/MoveHistory/EventTemplates/index.js
index 1d354fb009..795708b7a0 100644
--- a/src/constants/MoveHistory/EventTemplates/index.js
+++ b/src/constants/MoveHistory/EventTemplates/index.js
@@ -45,6 +45,7 @@ export { default as updateMTOShipmentStatus } from './UpdateMTOShipmentStatus/up
 export { default as updateMTOServiceItemMoveStatus } from './CreateMTOServiceItem/createMTOServiceItemUpdateMoveStatus';
 export { default as updateMoveEstimatedExcessWeight } from './UpdateMTOShipment/updateMTOShipmentPrimeEstimatedExcessWeight';
 export { default as updateMTOStatusServiceCounselingCompleted } from './UpdateMTOStatusServiceCounselingCompleted/updateMTOStatusServiceCounselingCompleted';
+export { default as updateMTOStatusServiceCounselingCompletedPPM } from './UpdateMTOStatusServiceCounselingCompleted/updateMTOStatusServiceCounselingCompletedPPM';
 export { default as updateAllowanceByCounselor } from './UpdateAllowances/updateAllowanceByCounselor';
 export { default as updateServiceMemberByCounselor } from './UpdateServiceMember/updateServiceMemberByCounselor';
 export { default as updateServiceMemberByTOO } from './UpdateServiceMember/updateServiceMemberByTOO';
@@ -67,3 +68,21 @@ export { default as updateMovingExpense } from './UpdateMovingExpense/UpdateMovi
 export { default as updateWeightTicketProGear } from './UpdateWeightTicket/updateWeightTicketProGear';
 export { default as createMTOShipmentPPMDetails } from './CreateMTOShipment/createMTOShipmentPPMDetails';
 export { default as deleteShipmentPPM } from './DeleteShipment/DeleteShipmentPPM';
+export { default as updateCloseoutOffice } from './UpdateCloseoutOffice/updateCloseoutOffice';
+export { default as createMTOShipmentUpdateMoves } from './CreateMTOShipment/CreateMTOShipmentUpdateMoves';
+export { default as createWeightTicket } from './CreateWeightTicket/CreateWeightTicket';
+export { default as updateWeightTicketPPMShipments } from './UpdateWeightTicket/UpdateWeightTicketPPMShipments';
+export { default as deleteWeightTicket } from './DeleteWeightTicket/DeleteWeightTicket';
+export { default as deleteWeightTicketPPMShipments } from './DeleteWeightTicket/DeleteWeightTicketPPMShipments';
+export { default as createPPMUpload } from './CreateUpload/CreatePPMUpload';
+export { default as createProGearWeightTicket } from './CreateProGearWeightTicket/CreateProGearWeightTicket';
+export { default as deleteProGearWeightTicket } from './DeleteProGearWeightTicket/DeleteProGearWeightTicket';
+export { default as deleteWeightTicketUserUploads } from './DeleteWeightTicket/DeleteWeightTicketUserUploads';
+export { default as deleteProGearWeightTicketUserUploads } from './DeleteProGearWeightTicket/DeleteProGearWeightTicketUserUploads';
+export { default as createMovingExpense } from './CreateMovingExpense/CreateMovingExpense';
+export { default as deleteMovingExpense } from './DeleteMovingExpense/DeleteMovingExpense';
+export { default as deleteMovingExpenseUserUploads } from './DeleteMovingExpense/DeleteMovingExpenseUserUploads';
+export { default as submitMoveForApprovalMTOShipments } from './SubmitMoveForApproval/SubmitMoveForApprovalMTOShipments';
+export { default as submitMoveForApprovalPPMShipments } from './SubmitMoveForApproval/SubmitMoveForApprovalPPMShipments';
+export { default as submitPPMShipmentDocumentation } from './SubmitPPMShipmentDocumentation/SubmitPPMShipmentDocumentation';
+export { default as finishDocumentReview } from './FinishDocumentReview/finishDocumentReview';
diff --git a/src/constants/MoveHistory/UIDisplay/Operations.js b/src/constants/MoveHistory/UIDisplay/Operations.js
index c4d526de76..ba001eb49e 100644
--- a/src/constants/MoveHistory/UIDisplay/Operations.js
+++ b/src/constants/MoveHistory/UIDisplay/Operations.js
@@ -49,4 +49,14 @@ export default {
   updateProGearWeightTicket: 'updateProGearWeightTicket', // ghc.yaml, internal.yaml
   updateMovingExpense: 'updateMovingExpense', // ghc.yaml, internal.yaml
   deleteMTOShipment: 'deleteMTOShipment', // prime.yaml
+  updateCloseoutOffice: 'updateCloseoutOffice', // ghc.yaml, internal.yaml
+  createWeightTicket: 'createWeightTicket', // internal.yaml
+  deleteWeightTicket: 'deleteWeightTicket', // internal.yaml
+  createPPMUpload: 'createPPMUpload', // internal.yaml
+  createProGearWeightTicket: 'createProGearWeightTicket', // internal.yaml
+  deleteProGearWeightTicket: 'deleteProGearWeightTicket', // internal.yaml
+  createMovingExpense: 'createMovingExpense', // internal.yaml
+  deleteMovingExpense: 'deleteMovingExpense', // internal.yaml
+  submitPPMShipmentDocumentation: 'submitPPMShipmentDocumentation', // internal.yaml
+  finishDocumentReview: 'finishDocumentReview', // ghc.yaml
 };
diff --git a/src/pages/Office/MoveHistory/LabeledDetails.jsx b/src/pages/Office/MoveHistory/LabeledDetails.jsx
index f9065d5b7a..30d8a3ae65 100644
--- a/src/pages/Office/MoveHistory/LabeledDetails.jsx
+++ b/src/pages/Office/MoveHistory/LabeledDetails.jsx
@@ -1,3 +1,4 @@
+/* eslint-disable camelcase */
 import React from 'react';

 import styles from './LabeledDetails.module.scss';
@@ -5,91 +6,145 @@ import styles from './LabeledDetails.module.scss';
 import booleanFields from 'constants/MoveHistory/Database/BooleanFields';
 import dateFields from 'constants/MoveHistory/Database/DateFields';
 import fieldMappings from 'constants/MoveHistory/Database/FieldMappings';
+import distanceFields from 'constants/MoveHistory/Database/DistanceFields';
+import timeUnitFields from 'constants/MoveHistory/Database/TimeUnitFields';
 import weightFields from 'constants/MoveHistory/Database/WeightFields';
 import monetaryFields from 'constants/MoveHistory/Database/MonetaryFields';
 import { shipmentTypes } from 'constants/shipments';
 import { HistoryLogRecordShape } from 'constants/MoveHistory/UIDisplay/HistoryLogShape';
 import optionFields from 'constants/MoveHistory/Database/OptionFields';
+import statusFields from 'constants/MoveHistory/Database/StatusFields';
 import {
   formatCents,
   formatCustomerDate,
+  formatDistanceUnitMiles,
+  formatTimeUnitDays,
   formatWeight,
   formatYesNoMoveHistoryValue,
   toDollarString,
 } from 'utils/formatters';

-const retrieveTextToDisplay = (fieldName, value) => {
-  const emptyValue = '—';
+export const withMappings = () => {
+  const self = {
+    displayMappings: [],
+  };
+
+  const getResult = ([mapping, fn] = []) => ({
+    mapping,
+    fn,
+  });
+
+  const defaultField = [
+    {},
+    ({ value }) =>
+      (value in optionFields && optionFields[value]) ||
+      (value in statusFields && statusFields[value]) ||
+      (`${value}` && value) ||
+      '—',
+  ];
+
+  self.addNameMappings = (mappings) => {
+    self.displayMappings = self.displayMappings.concat(mappings);
+    return self;
+  };
+  self.getMappedDisplayName = (field) =>
+    getResult(self.displayMappings.find(([mapping]) => field in mapping) || defaultField);
+  return self;
+};
+
+export const { displayMappings, getMappedDisplayName } = withMappings().addNameMappings([
+  [weightFields, ({ value }) => formatWeight(Number(value))],
+  [dateFields, ({ value }) => formatCustomerDate(value)],
+  [booleanFields, ({ value }) => formatYesNoMoveHistoryValue(value)],
+  [monetaryFields, ({ value }) => toDollarString(formatCents(value))],
+  [timeUnitFields, ({ value }) => formatTimeUnitDays(value)],
+  [distanceFields, ({ value }) => formatDistanceUnitMiles(value)],
+  [statusFields, ({ value }) => statusFields[value]],
+  [optionFields, ({ value }) => optionFields[value]],
+]);
+
+export const retrieveTextToDisplay = (fieldName, value) => {
   const displayName = fieldMappings[fieldName];
-  let displayValue = value;
-
-  if (displayName === fieldMappings.storage_in_transit) {
-    displayValue = `${displayValue} days`;
-  } else if (weightFields[fieldName]) {
-    // turn string value into number so it can be formatted correctly
-    displayValue = formatWeight(Number(displayValue));
-  } else if (optionFields[displayValue]) {
-    displayValue = optionFields[displayValue];
-  } else if (dateFields[fieldName]) {
-    displayValue = formatCustomerDate(displayValue);
-  } else if (booleanFields[fieldName]) {
-    displayValue = formatYesNoMoveHistoryValue(displayValue);
-  } else if (monetaryFields[fieldName]) {
-    displayValue = toDollarString(formatCents(displayValue));
-  }
-
-  if (!displayValue) {
-    displayValue = emptyValue;
-  }
+
+  const { fn: valueFormatFn } = getMappedDisplayName(fieldName);
+  const displayValue = valueFormatFn({ value });

   return {
     displayName,
-    displayValue,
+    displayValue: (!`${value}` && '—') || (value && displayValue) || '—',
   };
 };

+// testable for code coverage //
+export const createLineItemLabel = (shipmentType, shipmentIdDisplay, serviceItemName, movingExpenseType) =>
+  [shipmentType && `${shipmentTypes[shipmentType]} shipment #${shipmentIdDisplay}`, serviceItemName, movingExpenseType]
+    .filter((e) => e)
+    .join(', ');
+
+// testable for code coverage //
+
+// Filter out empty values unless they used to be non-empty
+// These values may be non-nullish in oldValues and nullish in changedValues
+// Use the existing keys in changed or old values to check against keys listed in fieldMappings
+export const filterInLineItemValues = (changedValues, oldValues) =>
+  Object.entries({ ...oldValues, ...changedValues }).filter(([theField]) => {
+    if (!(fieldMappings[theField]?.length >= 0)) return false;
+
+    const changed = changedValues || {};
+    const old = oldValues || {};
+
+    const isInChangedValues = theField in changedValues;
+
+    if (isInChangedValues)
+      switch (changed[theField]) {
+        case undefined:
+        case null:
+          break;
+        default:
+          if (changed[theField] === '' && !(theField in old)) return false;
+          return true;
+      }
+
+    if (isInChangedValues)
+      switch (old[theField]) {
+        case undefined:
+        case null:
+        case '':
+          break;
+        default:
+          return true;
+      }
+
+    return false;
+  });
+
 const LabeledDetails = ({ historyRecord }) => {
-  const changedValuesToUse = historyRecord.changedValues;
-  let shipmentDisplay = '';
+  const { changedValues, oldValues = {} } = historyRecord;
+
+  const { shipment_type, shipment_id_display, service_item_name, moving_expense_type, ...changedValuesToUse } =
+    changedValues;

   // Check for shipment_type to use it as a header for the row
-  if ('shipment_type' in changedValuesToUse) {
-    shipmentDisplay = shipmentTypes[changedValuesToUse.shipment_type];
-    shipmentDisplay += ` shipment #${changedValuesToUse.shipment_id_display}`;
-    delete changedValuesToUse.shipment_type;
-  }
-
-  if ('service_item_name' in changedValuesToUse) {
-    shipmentDisplay += `, ${changedValuesToUse.service_item_name}`;
-    delete changedValuesToUse.service_item_name;
-  }
-
-  if ('moving_expense_type' in changedValuesToUse) {
-    shipmentDisplay += `, ${changedValuesToUse.moving_expense_type}`;
-    delete changedValuesToUse.moving_expense_type;
-  }
-
-  /* Filter out empty values unless they used to be non-empty
-     These values may be non-nullish in oldValues and nullish in changedValues */
-  const dbFieldsToDisplay = Object.keys(fieldMappings).filter((dbField) => {
+  const shipmentDisplay = createLineItemLabel(
+    shipment_type,
+    shipment_id_display,
+    service_item_name,
+    moving_expense_type,
+  );
+
+  const lineItems = filterInLineItemValues(changedValuesToUse, oldValues).map(([label, value]) => {
+    const { displayName, displayValue } = retrieveTextToDisplay(label, value);
     return (
-      changedValuesToUse[dbField] ||
-      (dbField in changedValuesToUse && historyRecord.oldValues && historyRecord.oldValues[dbField])
+      <div key={label}>
+        <b>{displayName}</b>: {displayValue}
+      </div>
     );
   });

   return (
     <>
       <span className={styles.shipmentType}>{shipmentDisplay}</span>
-      {dbFieldsToDisplay.map((modelField) => {
-        const { displayName, displayValue } = retrieveTextToDisplay(modelField, changedValuesToUse[modelField]);
-
-        return (
-          <div key={modelField}>
-            <b>{displayName}</b>: {displayValue}
-          </div>
-        );
-      })}
+      {lineItems}
     </>
   );
 };
diff --git a/src/pages/Office/MoveHistory/LabeledDetails.test.jsx b/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
index 7e6ee63d27..a99baa4020 100644
--- a/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
+++ b/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
@@ -1,116 +1,61 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';

-import LabeledDetails from './LabeledDetails';
+import LabeledDetails, { retrieveTextToDisplay } from './LabeledDetails';

 import { SHIPMENT_OPTIONS } from 'shared/constants';
-import {
-  formatCustomerDate,
-  formatEvaluationReportLocation,
-  formatWeight,
-  formatYesNoMoveHistoryValue,
-  toDollarString,
-} from 'utils/formatters';
-import * as fieldDefault from 'constants/MoveHistory/Database/FieldMappings';
-
-const FieldMappings = fieldDefault.default;

 describe('LabeledDetails', () => {
   describe('for each changed value', () => {
-    const historyRecord = {
-      changedValues: {
-        customer_remarks: 'Test customer remarks',
-        counselor_remarks: 'Test counselor remarks',
-        billable_weight_justification: 'Test TIO remarks',
-        billable_weight_cap: '400',
-        tac_type: 'HHG',
-        sac_type: 'NTS',
-        service_order_number: '1234',
-        authorized_weight: '500',
-        storage_in_transit: '5',
-        dependents_authorized: true,
-        pro_gear_weight: '100',
-        pro_gear_weight_spouse: '50',
-        required_medical_equipment_weight: '300',
-        organizational_clothing_and_individual_equipment: 'false',
-        orders_type: 'PERMANENT_CHANGE_OF_STATION',
-        orders_type_detail: 'HHG_PERMITTED',
-        origin_duty_location_name: 'Origin duty location name',
-        new_duty_location_name: 'New duty location name',
-        orders_number: '1111',
-        tac: '2222',
-        sac: '4444',
-        nts_tac: '3333',
-        nts_sac: '5555',
-        department_indicator: 'AIR_AND_SPACE_FORCE',
-        grade: 'E_1',
-        actual_pickup_date: '2022-01-01',
-        prime_actual_weight: '100 lbs',
-        destination_address_type: 'HOME_OF_SELECTION',
-        affiliation: 'COAST_GUARD',
-        requested_delivery_date: '2023-02-05',
-        sit_entry_date: '2023-04-05',
-        sit_departure_date: '2023-03-05',
-        sit_expected: true,
-        sit_location: 'Destination',
-        sit_estimated_weight: '500',
-        sit_estimated_cost: '120000',
-        estimated_incentive: '850',
-        shipment_weight: '100',
-      },
+    const changedValues = {
+      customer_remarks: 'Test customer remarks',
+      counselor_remarks: 'Test counselor remarks',
+      billable_weight_justification: 'Test TIO remarks',
+      billable_weight_cap: '400',
+      tac_type: 'HHG',
+      sac_type: 'NTS',
+      service_order_number: '1234',
+      authorized_weight: '500',
+      storage_in_transit: '5',
+      dependents_authorized: true,
+      pro_gear_weight: '100',
+      pro_gear_weight_spouse: '50',
+      required_medical_equipment_weight: '300',
+      organizational_clothing_and_individual_equipment: 'false',
+      orders_type: 'PERMANENT_CHANGE_OF_STATION',
+      orders_type_detail: 'HHG_PERMITTED',
+      origin_duty_location_name: 'Origin duty location name',
+      new_duty_location_name: 'New duty location name',
+      orders_number: '1111',
+      tac: '2222',
+      sac: '4444',
+      nts_tac: '3333',
+      nts_sac: '5555',
+      department_indicator: 'AIR_AND_SPACE_FORCE',
+      grade: 'E_1',
+      actual_pickup_date: '2022-01-01',
+      prime_actual_weight: '100 lbs',
+      destination_address_type: 'HOME_OF_SELECTION',
+      affiliation: 'COAST_GUARD',
+      requested_delivery_date: '2023-02-05',
+      sit_entry_date: '2023-04-05',
+      sit_departure_date: '2023-03-05',
+      sit_expected: true,
+      sit_location: 'Destination',
+      sit_estimated_weight: '500',
+      sit_estimated_cost: '120000',
+      estimated_incentive: '850',
+      shipment_weight: '100',
     };

-    const testCases = [
-      [FieldMappings.counselor_remarks, ': Test counselor remarks'],
-      [FieldMappings.customer_remarks, ': Test customer remarks'],
-      [FieldMappings.billable_weight_justification, ': Test TIO remarks'],
-      [FieldMappings.billable_weight_cap, ': 400 lbs'],
-      [FieldMappings.tac_type, ': HHG'],
-      [FieldMappings.sac_type, ': NTS'],
-      [FieldMappings.service_order_number, ': 1234'],
-      [FieldMappings.authorized_weight, ': 500 lbs'],
-      [FieldMappings.storage_in_transit, ': 5 days'],
-      [FieldMappings.dependents_authorized, ': Yes'],
-      [FieldMappings.pro_gear_weight, ': 100 lbs'],
-      [FieldMappings.pro_gear_weight_spouse, ': 50 lbs'],
-      [FieldMappings.required_medical_equipment_weight, ': 300 lbs'],
-      [FieldMappings.orders_type, ': Permanent Change Of Station (PCS)'],
-      [FieldMappings.orders_type_detail, ': Shipment of HHG Permitted'],
-      [FieldMappings.origin_duty_location_name, ': Origin duty location name'],
-      [FieldMappings.new_duty_location_name, ': New duty location name'],
-      [FieldMappings.orders_number, ': 1111'],
-      [FieldMappings.tac, ': 2222'],
-      [FieldMappings.nts_tac, ': 3333'],
-      [FieldMappings.sac, ': 4444'],
-      [FieldMappings.nts_sac, ': 5555'],
-      [FieldMappings.grade, ': E-1'],
-      [FieldMappings.department_indicator, ': Air Force and Space Force'],
-      [FieldMappings.actual_pickup_date, `: ${formatCustomerDate(historyRecord.changedValues.actual_pickup_date)}`],
-      [FieldMappings.shipment_weight, ': 100 lbs'],
-      [FieldMappings.destination_address_type, ': Home of selection (HOS)'],
-      [
-        FieldMappings.requested_delivery_date,
-        `: ${formatCustomerDate(historyRecord.changedValues.requested_delivery_date)}`,
-      ],
-      [FieldMappings.sit_entry_date, `: ${formatCustomerDate(historyRecord.changedValues.sit_entry_date)}`],
-      [FieldMappings.sit_departure_date, `: ${formatCustomerDate(historyRecord.changedValues.sit_departure_date)}`],
-      [FieldMappings.sit_expected, `: ${formatYesNoMoveHistoryValue(historyRecord.changedValues.sit_expected)}`],
-      [
-        FieldMappings.sit_location,
-        `: ${formatEvaluationReportLocation(historyRecord.changedValues.sit_location.toUpperCase())}`,
-      ],
-      [FieldMappings.sit_estimated_weight, `: ${formatWeight(historyRecord.changedValues.sit_estimated_weight)}`],
-      [FieldMappings.sit_estimated_cost, `: ${toDollarString(historyRecord.changedValues.sit_estimated_cost / 100)}`],
-      [FieldMappings.estimated_incentive, `: ${toDollarString(historyRecord.changedValues.estimated_incentive / 100)}`],
-    ];
-
-    it.each(testCases)('it renders %s%s', (displayName, value) => {
-      render(<LabeledDetails historyRecord={historyRecord} />);
-      const displayingElements = screen.getAllByText(displayName);
-      const displayingElement = displayingElements.find((element) => element.parentElement.textContent.includes(value));
-      const parent = displayingElement.parentElement;
-      expect(parent.textContent).toContain(displayName);
-      expect(parent.textContent).toContain(value);
+    const testCases = Object.entries(changedValues).map(([fieldName, value]) => {
+      const { displayName, displayValue } = retrieveTextToDisplay(fieldName, value);
+      return [fieldName, value, displayName, displayValue];
+    });
+
+    it.each(testCases)("it renders [%s %s] as '%s: %s'", (fieldName, value, displayName, displayValue) => {
+      const { baseElement } = render(<LabeledDetails historyRecord={{ changedValues: { [fieldName]: value } }} />);
+      expect(baseElement).toHaveTextContent(`${displayName}: ${displayValue}`, { normalizeWhitespace: true });
     });
   });

@@ -163,10 +108,8 @@ it('does render text for changed values that are blank when they exist in the ol
     },
   };

-  render(<LabeledDetails historyRecord={historyRecord} />);
-
-  expect(screen.getByText('Counselor remarks')).toBeInTheDocument();
-  expect(screen.getByText('—', { exact: false })).toBeInTheDocument();
+  const { baseElement } = render(<LabeledDetails historyRecord={historyRecord} />);

-  expect(await screen.queryByText('These remarks were deleted')).not.toBeInTheDocument();
+  expect(baseElement).toHaveTextContent('Counselor remarks');
+  expect(baseElement).toHaveTextContent('—');
 });
diff --git a/src/utils/formatPPMData.js b/src/utils/formatPPMData.js
new file mode 100644
index 0000000000..0855dbafaa
--- /dev/null
+++ b/src/utils/formatPPMData.js
@@ -0,0 +1,47 @@
+/* eslint-disable camelcase */
+import { formatMoveHistoryFullAddressFromJSON } from './formatters';
+
+import { PPM_UPLOAD_TYPES, PPM_UPLOAD_TYPES_LABELS } from 'constants/ppmUploadTypes';
+
+const formatDisplayName = (uploadType) => {
+  if (PPM_UPLOAD_TYPES.some((type) => type === uploadType)) {
+    return PPM_UPLOAD_TYPES_LABELS[uploadType];
+  }
+
+  return '';
+};
+
+export const formatCloseoutOfficeFor = ({ changedValues = {}, context: [{ closeout_office_name }] = [{}] }) =>
+  'closeout_office_id' in changedValues &&
+  closeout_office_name && { closeout_office_name: changedValues.closeout_office_id && closeout_office_name };
+
+export const formatUploadTypeFor = ({ context: [{ upload_type }] = [{}] }) =>
+  upload_type && { upload_type: formatDisplayName(upload_type) };
+
+export const formatFileNameFor = ({ context: [{ filename }] = [{}] }) => filename && { filename };
+
+export const formatW2AddressFor = ({ changedValues: { w2_address_id } = {}, context: [{ w2_address }] = [{}] }) =>
+  w2_address_id && w2_address && { w2_address: formatMoveHistoryFullAddressFromJSON(w2_address) };
+
+export function formatDataForPPM(historyRecord) {
+  const ppmValues = {
+    ...formatCloseoutOfficeFor(historyRecord),
+    ...formatUploadTypeFor(historyRecord),
+    ...formatFileNameFor(historyRecord),
+    ...formatW2AddressFor(historyRecord),
+  };
+
+  // it was requested that we add a status of 'ADDED' once a customer finishes creating a ppm
+  // will require refactor down the line but this has been approved for now
+  if (
+    historyRecord.changedValues?.has_requested_advance !== undefined &&
+    historyRecord.oldValues?.has_requested_advance === null &&
+    historyRecord.oldValues?.advance_amount_requested === null
+  ) {
+    ppmValues.ppm_status = 'ADDED';
+  }
+
+  return ppmValues;
+}
+
+export default { formatDataForPPM };
diff --git a/src/utils/formatters.js b/src/utils/formatters.js
index d41a2dfbd3..19cb9cd7b6 100644
--- a/src/utils/formatters.js
+++ b/src/utils/formatters.js
@@ -212,6 +212,10 @@ export const formatMoveHistoryFullAddress = (address) => {
   return formattedAddress;
 };

+export const formatMoveHistoryFullAddressFromJSON = (address) => {
+  return formatMoveHistoryFullAddress(JSON.parse(address));
+};
+
 export const formatMoveHistoryAgent = (agent) => {
   let agentLabel = '';
   if (agent.agent_type === 'RECEIVING_AGENT') {
@@ -479,3 +483,11 @@ export function formatEvaluationReportLocation(location) {
       return undefined;
   }
 }
+
+export function formatTimeUnitDays(days) {
+  return `${days} days`;
+}
+
+export function formatDistanceUnitMiles(distance) {
+  return `${distance} miles`;
+}
