diff --git a/src/pages/Office/MoveHistory/MoveHistory.test.jsx b/src/pages/Office/MoveHistory/MoveHistory.test.jsx
index 87d8ee316e..85094b183c 100644
--- a/src/pages/Office/MoveHistory/MoveHistory.test.jsx
+++ b/src/pages/Office/MoveHistory/MoveHistory.test.jsx
@@ -5,6 +5,8 @@ import MoveHistory from './MoveHistory';

 import { MockProviders } from 'testUtils';

+const dateTimeFormatRegex = /(\d{2}) (\w{3}) (\d{2}) (\d{2}):(\d{2})/;
+
 const mockMoveCode = 'DVRS0N';

 jest.mock('hooks/queries', () => ({
@@ -135,12 +137,12 @@ describe('MoveHistory', () => {
       expect(screen.getByText('Move history (2)')).toBeInTheDocument();
       expect(screen.getByRole('table')).toBeInTheDocument();

-      expect(screen.getByTestId('move-history-date-time-0')).toHaveTextContent('09 Mar 22 15:33');
+      expect(screen.getByTestId('move-history-date-time-0')).toHaveTextContent(dateTimeFormatRegex);
       expect(screen.getByTestId('move-history-event-0')).toHaveTextContent('Approved move');
       expect(screen.getByTestId('move-history-details-0')).toBeInTheDocument();
       expect(screen.getByTestId('move-history-modified-by-0')).toBeInTheDocument();

-      expect(screen.getByTestId('move-history-date-time-1')).toHaveTextContent('08 Mar 22 18:28');
+      expect(screen.getByTestId('move-history-date-time-1')).toHaveTextContent(dateTimeFormatRegex);
       expect(screen.getByTestId('move-history-event-1')).toHaveTextContent('Approved shipment');
       expect(screen.getByTestId('move-history-details-1')).toBeInTheDocument();
       expect(screen.getByTestId('move-history-modified-by-1')).toBeInTheDocument();
