diff --git a/src/hooks/queries.js b/src/hooks/queries.js
index a057edc14c..34e7538b5e 100644
--- a/src/hooks/queries.js
+++ b/src/hooks/queries.js
@@ -30,7 +30,7 @@ import {
   getServicesCounselingPPMQueue,
 } from 'services/ghcApi';
 import { getLoggedInUserQueries } from 'services/internalApi';
-import { getPrimeSimulatorAvailableMoves, getPrimeSimulatorMove } from 'services/primeApi';
+import { getPrimeSimulatorMove } from 'services/primeApi';
 import { getQueriesStatus } from 'utils/api';
 import {
   PAYMENT_REQUESTS,
@@ -47,7 +47,6 @@ import {
   CUSTOMER,
   SERVICES_COUNSELING_QUEUE,
   SHIPMENTS_PAYMENT_SIT_BALANCE,
-  PRIME_SIMULATOR_AVAILABLE_MOVES,
   PRIME_SIMULATOR_MOVE,
   CUSTOMER_SUPPORT_REMARKS,
   QAE_CSR_MOVE_SEARCH,
@@ -766,34 +765,6 @@ export const useMoveDetailsQueries = (moveCode) => {
   };
 };

-const usePrimeSimulatorAvailableMovesQueries = (dateValue) => {
-  const { data = {}, ...primeSimulatorAvailableMovesQuery } = useQuery(
-    // [PRIME_SIMULATOR_AVAILABLE_MOVES, {}],
-    [PRIME_SIMULATOR_AVAILABLE_MOVES, `${dateValue}`],
-    ({ queryKey: [key, date] }) => {
-      return getPrimeSimulatorAvailableMoves(key);
-    },
-  );
-  const { isLoading, isError, isSuccess } = getQueriesStatus([primeSimulatorAvailableMovesQuery]);
-  // README: This queueResult is being artificially constructed rather than
-  // created using the `..dataProp` destructering of other functions because
-  // the Prime API does not return an Object that the TableQueue component can
-  // consume. So the queueResult mimics that Objects properties since `data` in
-  // this case is a simple Array of Prime Available Moves.
-  const queueResult = {
-    data,
-    page: 1,
-    perPage: data.length,
-    totalCount: data.length,
-  };
-  return {
-    queueResult,
-    isLoading,
-    isError,
-    isSuccess,
-  };
-};
-
 export const usePrimeSimulatorGetMove = (moveCode) => {
   const { data: moveTaskOrder, ...primeSimulatorGetMoveQuery } = useQuery(
     [PRIME_SIMULATOR_MOVE, moveCode],
diff --git a/src/pages/PrimeUI/AvailableMoves/AvailableMovesQueue.jsx b/src/pages/PrimeUI/AvailableMoves/AvailableMovesQueue.jsx
index 252b9ce4b8..45949edc86 100644
--- a/src/pages/PrimeUI/AvailableMoves/AvailableMovesQueue.jsx
+++ b/src/pages/PrimeUI/AvailableMoves/AvailableMovesQueue.jsx
@@ -64,9 +64,8 @@ const PrimeSimulatorAvailableMoves = () => {
       return getPrimeSimulatorAvailableMoves(key, date);
     },
   );
-
   // eslint-disable-next-line no-use-before-define
-  const apiQuery = (...items) => {
+  const apiQuery = () => {
     const { isLoading, isError, isSuccess } = getQueriesStatus([primeSimulatorAvailableMovesQuery]);
     // README: This queueResult is being artificially constructed rather than
     // created using the `..dataProp` destructering of other functions because
