diff --git a/src/constants/MoveHistory/Database/FieldMappings.js b/src/constants/MoveHistory/Database/FieldMappings.js
index 2babc48659..3fefe61106 100644
--- a/src/constants/MoveHistory/Database/FieldMappings.js
+++ b/src/constants/MoveHistory/Database/FieldMappings.js
@@ -110,12 +110,12 @@ export default {
   net_weight_remarks: 'Remarks',
   adjusted_net_weight: 'Net weight',
   amount: 'Amount',
-  sit_start_date: 'SIT Start Date',
-  sit_end_date: 'SIT End Date',
-  weight: 'Weight',
+  sit_start_date: 'SIT start date',
+  sit_end_date: 'SIT end date',
+  shipment_weight: 'Shipment weight',
   belongs_to_self: 'Service member',
   sit_customer_contacted: 'Customer contacted',
-  sit_location: 'SIT Location',
-  sit_estimated_cost: 'SIT Estimated Cost',
-  estimated_incentive: 'PPM Estimated Incentive',
+  sit_location: 'SIT location',
+  sit_estimated_cost: 'SIT estimated cost',
+  estimated_incentive: 'PPM estimated incentive',
 };
diff --git a/src/constants/MoveHistory/Database/MonetaryFields.js b/src/constants/MoveHistory/Database/MonetaryFields.js
index b8e412f0d8..d5ee4fff37 100644
--- a/src/constants/MoveHistory/Database/MonetaryFields.js
+++ b/src/constants/MoveHistory/Database/MonetaryFields.js
@@ -1,6 +1,6 @@
 export default {
   advance_amount_requested: 'advance_amount_requested',
   amount: 'amount',
-  sit_estimated_cost: 'SIT Estimated Cost',
-  estimated_incentive: 'PPM Estimated Incentive',
+  sit_estimated_cost: 'SIT estimated cost',
+  estimated_incentive: 'PPM estimated incentive',
 };
diff --git a/src/constants/MoveHistory/Database/WeightFields.js b/src/constants/MoveHistory/Database/WeightFields.js
index 7cd10d6c5b..e0df577a94 100644
--- a/src/constants/MoveHistory/Database/WeightFields.js
+++ b/src/constants/MoveHistory/Database/WeightFields.js
@@ -15,4 +15,5 @@ export default {
   adjusted_net_weight: 'adjusted_net_weight',
   empty_weight: 'empty_weight',
   full_weight: 'full_weight',
+  shipment_weight: 'shipment_weight',
 };
diff --git a/src/pages/Office/MoveHistory/LabeledDetails.test.jsx b/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
index cbfbb0a74f..aab478c8da 100644
--- a/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
+++ b/src/pages/Office/MoveHistory/LabeledDetails.test.jsx
@@ -4,6 +4,17 @@ import { render, screen } from '@testing-library/react';
 import LabeledDetails from './LabeledDetails';

 import { SHIPMENT_OPTIONS } from 'shared/constants';
+import {
+  formatCustomerDate,
+  formatDate,
+  formatEvaluationReportLocation,
+  formatWeight,
+  formatYesNoMoveHistoryValue,
+  toDollarString,
+} from 'utils/formatters';
+import * as fieldDefault from 'constants/MoveHistory/Database/FieldMappings';
+
+const FieldMappings = fieldDefault.default;

 describe('LabeledDetails', () => {
   describe('for each changed value', () => {
@@ -29,8 +40,8 @@ describe('LabeledDetails', () => {
         new_duty_location_name: 'New duty location name',
         orders_number: '1111',
         tac: '2222',
-        sac: '3333',
-        nts_tac: '4444',
+        sac: '4444',
+        nts_tac: '3333',
         nts_sac: '5555',
         department_indicator: 'AIR_AND_SPACE_FORCE',
         grade: 'E_1',
@@ -38,43 +49,70 @@ describe('LabeledDetails', () => {
         prime_actual_weight: '100 lbs',
         destination_address_type: 'HOME_OF_SELECTION',
         affiliation: 'COAST_GUARD',
+        requested_delivery_date: '2023-02-05',
+        sit_entry_date: '2023-04-05',
+        sit_departure_date: '2023-03-05',
+        sit_expected: true,
+        sit_location: 'Destination',
+        sit_estimated_weight: '500',
+        sit_estimated_cost: '120000',
+        estimated_incentive: '850',
+        shipment_weight: '100',
       },
     };
-    it.each([
-      ['Customer remarks', ': Test customer remarks'],
-      ['Counselor remarks', ': Test counselor remarks'],
-      ['Billable weight remarks', ': Test TIO remarks'],
-      ['Billable weight', ': 400 lbs'],
-      ['TAC type', ': HHG'],
-      ['SAC type', ': NTS'],
-      ['Service order number', ': 1234'],
-      ['Authorized weight', ': 500 lbs'],
-      ['Storage in transit (SIT)', ': 5 days'],
-      ['Dependents', ': Yes'],
-      ['Pro-gear weight', ': 100 lbs'],
-      ['Spouse pro-gear weight', ': 50 lbs'],
-      ['Required medical equipment', ': 300 lbs'],
-      ['Orders type', ': Permanent Change Of Station (PCS)'],
-      ['Orders type detail', ': Shipment of HHG Permitted'],
-      ['Origin duty location name', ': Origin duty location name'],
-      ['New duty location name', ': New duty location name'],
-      ['Orders number', ': 1111'],
-      ['HHG TAC', ': 2222'],
-      ['NTS TAC', ': 3333'],
-      ['HHG SAC', ': 4444'],
-      ['NTS SAC', ': 5555'],
-      ['Pay grade', ': E-1'],
-      ['Dept. indicator', ': Air Force and Space Force'],
-      ['Departure date', ': 01 Jan 2022'],
-      ['Shipment weight', ': 100 lbs'],
-      ['Destination type', ': Home of selection (HOS)'],
-      ['Branch', ': Coast Guard'],
-    ])('it renders %s%s', (displayName, value) => {
-      render(<LabeledDetails historyRecord={historyRecord} />);
-
-      expect(screen.getByText(displayName)).toBeInTheDocument();

-      expect(screen.getByText(value)).toBeInTheDocument();
+    const testCases = [
+      [FieldMappings.counselor_remarks, ': Test counselor remarks'],
+      [FieldMappings.customer_remarks, ': Test customer remarks'],
+      [FieldMappings.billable_weight_justification, ': Test TIO remarks'],
+      [FieldMappings.billable_weight_cap, ': 400 lbs'],
+      [FieldMappings.tac_type, ': HHG'],
+      [FieldMappings.sac_type, ': NTS'],
+      [FieldMappings.service_order_number, ': 1234'],
+      [FieldMappings.authorized_weight, ': 500 lbs'],
+      [FieldMappings.storage_in_transit, ': 5 days'],
+      [FieldMappings.dependents_authorized, ': Yes'],
+      [FieldMappings.pro_gear_weight, ': 100 lbs'],
+      [FieldMappings.pro_gear_weight_spouse, ': 50 lbs'],
+      [FieldMappings.required_medical_equipment_weight, ': 300 lbs'],
+      [FieldMappings.orders_type, ': Permanent Change Of Station (PCS)'],
+      [FieldMappings.orders_type_detail, ': Shipment of HHG Permitted'],
+      [FieldMappings.origin_duty_location_name, ': Origin duty location name'],
+      [FieldMappings.new_duty_location_name, ': New duty location name'],
+      [FieldMappings.orders_number, ': 1111'],
+      [FieldMappings.tac, ': 2222'],
+      [FieldMappings.nts_tac, ': 3333'],
+      [FieldMappings.sac, ': 4444'],
+      [FieldMappings.nts_sac, ': 5555'],
+      [FieldMappings.grade, ': E-1'],
+      [FieldMappings.department_indicator, ': Air Force and Space Force'],
+      [FieldMappings.actual_pickup_date, `: ${formatCustomerDate(historyRecord.changedValues.actual_pickup_date)}`],
+      [FieldMappings.shipment_weight, ': 100 lbs'],
+      [FieldMappings.destination_address_type, ': Home of selection (HOS)'],
+      [
+        FieldMappings.requested_delivery_date,
+        `: ${formatCustomerDate(historyRecord.changedValues.requested_delivery_date)}`,
+      ],
+      [FieldMappings.sit_entry_date, `: ${formatCustomerDate(historyRecord.changedValues.sit_entry_date)}`],
+      [FieldMappings.sit_departure_date, `: ${formatCustomerDate(historyRecord.changedValues.sit_departure_date)}`],
+      [FieldMappings.sit_expected, `: ${formatYesNoMoveHistoryValue(historyRecord.changedValues.sit_expected)}`],
+      [
+        FieldMappings.sit_location,
+        `: ${formatEvaluationReportLocation(historyRecord.changedValues.sit_location.toUpperCase())}`,
+      ],
+      [FieldMappings.sit_estimated_weight, `: ${formatWeight(historyRecord.changedValues.sit_estimated_weight)}`],
+      [FieldMappings.sit_estimated_cost, `: ${toDollarString(historyRecord.changedValues.sit_estimated_cost / 100)}`],
+      [FieldMappings.estimated_incentive, `: ${toDollarString(historyRecord.changedValues.estimated_incentive / 100)}`],
+    ];
+
+    it.each(testCases)('it renders %s%s', (displayName, value) => {
+      render(<LabeledDetails historyRecord={historyRecord} />);
+      const displayingElements = screen.getAllByText(displayName);
+      const texts = displayingElements.map((e) => e.parentElement.textContent);
+      const displayingElement = displayingElements.find((element) => element.parentElement.textContent.includes(value));
+      const parent = displayingElement.parentElement;
+      expect(parent.textContent).toContain(displayName);
+      expect(parent.textContent).toContain(value);
     });
   });

