diff --git a/src/components/PrimeUI/CreateShipmentServiceItemForm/CreateShipmentServiceItemForm.jsx b/src/components/PrimeUI/CreateShipmentServiceItemForm/CreateShipmentServiceItemForm.jsx
index c6c1d1fa68..d0a08869f7 100644
--- a/src/components/PrimeUI/CreateShipmentServiceItemForm/CreateShipmentServiceItemForm.jsx
+++ b/src/components/PrimeUI/CreateShipmentServiceItemForm/CreateShipmentServiceItemForm.jsx
@@ -7,20 +7,25 @@ import DestinationSITServiceItemForm from './DestinationSITServiceItemForm';
 import OriginSITServiceItemForm from './OriginSITServiceItemForm';
 import ShuttleSITServiceItemForm from './ShuttleSITServiceItemForm';
 import DomesticCratingForm from './DomesticCratingForm';
+import PPMAdanceRequestServiceItemForm from './PPMAdvanceRequestServiceItemForm';

 import { ShipmentShape } from 'types/shipment';
 import { createServiceItemModelTypes } from 'constants/prime';
 import Shipment from 'components/PrimeUI/Shipment/Shipment';

 const CreateShipmentServiceItemForm = ({ shipment, createServiceItemMutation }) => {
-  const { MTOServiceItemOriginSIT, MTOServiceItemDestSIT, MTOServiceItemShuttle, MTOServiceItemDomesticCrating } =
-    createServiceItemModelTypes;
+  const {
+    MTOServiceItemShipmentAdvanceRequest,
+    MTOServiceItemOriginSIT,
+    MTOServiceItemDestSIT,
+    MTOServiceItemShuttle,
+    MTOServiceItemDomesticCrating,
+  } = createServiceItemModelTypes;
   const [selectedServiceItemType, setSelectedServiceItemType] = useState(MTOServiceItemOriginSIT);

   const handleServiceItemTypeChange = (event) => {
     setSelectedServiceItemType(event.target.value);
   };
-
   return (
     <div className={styles.CreateShipmentServiceItemForm}>
       <Shipment shipment={shipment} />
@@ -31,6 +36,9 @@ const CreateShipmentServiceItemForm = ({ shipment, createServiceItemMutation })
           <option value={MTOServiceItemDestSIT}>Destination SIT</option>
           <option value={MTOServiceItemShuttle}>Shuttle</option>
           <option value={MTOServiceItemDomesticCrating}>Domestic Crating</option>
+          <option disabled={shipment.shipmentType !== 'PPM'} value={MTOServiceItemShipmentAdvanceRequest}>
+            PPM Advance Request
+          </option>
         </>
       </Dropdown>
       {selectedServiceItemType === MTOServiceItemOriginSIT && (
@@ -45,6 +53,9 @@ const CreateShipmentServiceItemForm = ({ shipment, createServiceItemMutation })
       {selectedServiceItemType === MTOServiceItemDomesticCrating && (
         <DomesticCratingForm shipment={shipment} submission={createServiceItemMutation} />
       )}
+      {selectedServiceItemType === MTOServiceItemShipmentAdvanceRequest && shipment.shipmentType === 'PPM' && (
+        <PPMAdanceRequestServiceItemForm shipment={shipment} />
+      )}
     </div>
   );
 };
diff --git a/src/constants/prime.js b/src/constants/prime.js
index f78e76414c..755b3d5fd4 100644
--- a/src/constants/prime.js
+++ b/src/constants/prime.js
@@ -6,6 +6,7 @@ export const createServiceItemModelTypes = {
   MTOServiceItemDestSIT: 'MTOServiceItemDestSIT',
   MTOServiceItemShuttle: 'MTOServiceItemShuttle',
   MTOServiceItemDomesticCrating: 'MTOServiceItemDomesticCrating',
+  MTOServiceItemShipmentAdvanceRequest: 'MTOServiceItemShipmentAdvanceRequest',
 };

 export const shuttleServiceItemCodeOptions = [
diff --git a/src/constants/routes.js b/src/constants/routes.js
index aff437d167..28dd80fd97 100644
--- a/src/constants/routes.js
+++ b/src/constants/routes.js
@@ -126,4 +126,5 @@ export const primeSimulatorRoutes = {
   UPLOAD_SERVICE_REQUEST_DOCUMENTS_PATH: `${BASE_PRIME_SIMULATOR_PATH}/mto-service-items/:mtoServiceItemId/upload`,
   SHIPMENT_UPDATE_ADDRESS_PATH: `${BASE_PRIME_SIMULATOR_PATH}/shipments/:shipmentId/addresses/update`,
   SHIPMENT_UPDATE_REWEIGH_PATH: `${BASE_PRIME_SIMULATOR_PATH}/shipments/:shipmentId/reweigh/:reweighId/update`,
+  SHIPMENT_PPM_ADVANCES_PATH: `${BASE_PRIME_SIMULATOR_PATH}/shipments/:shipmentId/advances`,
 };
diff --git a/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx b/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx
index c07a910f8c..109200aa44 100644
--- a/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx
+++ b/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx
@@ -331,7 +331,7 @@ const PrimeUIShipmentUpdate = ({ setFlashMessage }) => {
                 </div>
               )}
               <Button type="button" onClick={onCancelShipmentClick} className="usa-button usa-button-secondary">
-                Cancel Shipment
+                Cancel Shipment??
               </Button>
               <Formik
                 initialValues={initialValues}
diff --git a/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdatePPMForm.jsx b/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdatePPMForm.jsx
index adb344fd06..a54905e613 100644
--- a/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdatePPMForm.jsx
+++ b/src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdatePPMForm.jsx
@@ -17,6 +17,8 @@ const PrimeUIShipmentUpdatePPMForm = () => {
   const { values } = useFormikContext();
   const { sitExpected, hasProGear } = values.ppmShipment;

+  console.log(values);
+
   return (
     <SectionWrapper className={`${formStyles.formSection} ${styles.formSectionHeader}`}>
       <h2 className={styles.sectionHeader}>Dates</h2>
diff --git a/src/shared/Entities/modules/ppms.js b/src/shared/Entities/modules/ppms.js
index f9ffb5d856..1b2f16521d 100644
--- a/src/shared/Entities/modules/ppms.js
+++ b/src/shared/Entities/modules/ppms.js
@@ -9,7 +9,7 @@ import { formatDateForSwagger } from 'shared/dates';
 const approvePpmLabel = 'PPMs.approvePPM';
 export const downloadPPMAttachmentsLabel = 'PPMs.downloadAttachments';
 const loadPPMsLabel = 'office.loadPPMs';
-const updatePPMLabel = 'office.updatePPM';
+export const updatePPMLabel = 'office.updatePPM';
 const approveReimbursementLabel = 'office.approveReimbursement';

 export function approvePPM(personallyProcuredMoveId, personallyProcuredMoveApproveDate, label = approvePpmLabel) {
