diff --git a/.envrc b/.envrc
index b93a46c55f..4cd10dcccd 100644
--- a/.envrc
+++ b/.envrc
@@ -122,8 +122,8 @@ export DB_SSL_MODE=disable
 # Experimental feature flags, these will be replaced by the config/env/*.env files for live deployments
 # Multi Move feature flag
 # This is utilized if flipt config does not receive a URL
-export FEATURE_FLAG_MULTI_MOVE=true
-export FEATURE_FLAG_COUNSELOR_MOVE_CREATE=true
+# export FEATURE_FLAG_MULTI_MOVE=true
+# export FEATURE_FLAG_COUNSELOR_MOVE_CREATE=true

 # Okta.mil configuration

diff --git a/src/pages/Office/MoveHistory/LabeledDetails.jsx b/src/pages/Office/MoveHistory/LabeledDetails.jsx
index bf37e33feb..89a072ebb3 100644
--- a/src/pages/Office/MoveHistory/LabeledDetails.jsx
+++ b/src/pages/Office/MoveHistory/LabeledDetails.jsx
@@ -86,18 +86,25 @@ export const createLineItemLabel = (shipmentType, shipmentIdDisplay, serviceItem
 // Filter out empty values unless they used to be non-empty
 // These values may be non-nullish in oldValues and nullish in changedValues
 // Use the existing keys in changed or old values to check against keys listed in fieldMappings
-export const filterInLineItemValues = (changed, old) =>
-  Object.entries({ ...old, ...changed }).filter(
-    ([theField, theValue]) =>
-      !!(
+export const filterInLineItemValues = (changedValues, oldValues) =>
+  Object.entries({ ...oldValues, ...changedValues }).filter(([theField, theValue]) => {
+    const changed = changedValues || {};
+    const old = oldValues || {};
+    const fieldInOld = theField in old;
+    const fieldInChanged = theField in changed;
+    const valueIsBlank = theValue?.length === 0;
+
+    return !!(
+      (
         fieldMappings[theField] &&
         // if changeValues has theField while theValue is either blank or theField exists the old values
-        ((theField in changed && (`${theValue}` || theField in old)) ||
+        ((fieldInChanged && ((`${theValue}`.length > 0 && !!theValue) || fieldInOld)) ||
           // or theField exists in changeValues
           // and oldValues is empty or not empty
-          (theField in changed && (!`${old[theField]}` || `${theValue}`)))
-      ),
-  );
+          (fieldInChanged && !`${old[theField]}`))
+      ) /* || `${theValue}` */
+    );
+  });

 const LabeledDetails = ({ historyRecord }) => {
   const { changedValues, oldValues = {} } = historyRecord;
